app/models/course.rb

```diff
@@ -12,6 +12,20 @@ def self.all_hash
     }
   end
 
+  def notes_number_hash
+    Hash[
+      self.terms.zip Term.where(course_id: self.id).collect { |term| term.students.
+        collect { |s| s.tasks.collect { |task| task.notes } } .flatten.count }
```

Comment:
There is has_many relation between term and tasks so you could write something like `{ |term| term.tasks.collect { |task| task.notes } .flatten.count }`